### PR TITLE
Legend tuple handler improve

### PR DESCRIPTION
**Pull request by three-comrades**
Continuation of #2904 and #2034.

Rebased, added some more documentation and adapted the example a bit.

I'm not sure how to ceate the test image. I created it on Ubuntu 14.04. which uses version 2.5.2 of the Freetype library - do I need to compile+install 2.5.3, or even the exact, possibly patched Fedora version that was used to create the rest of the images?

### CODE REVIEW

**Comment by three-comrades** [comment]
Eh, sry, I fucked the initial push up. Hopefully it's okay now.

**Comment by tacaswell** [comment]
:+1: awesome

attn @mdboom re question about freetype.

**Comment by QuLogic** [comment]
To create consistent text output, you need to set `local_freetype = True` in [`setup.cfg`](https://github.com/matplotlib/matplotlib/blob/master/setup.cfg.template) before building to use the bundled FreeType instead of the system one.

**Comment by mdboom** [comment]
@QuLogic is exactly right about freetype here.  This is new since 1.5.x, so is documented in the "devdocs", not the default doc version, here:

http://matplotlib.org/devdocs/devel/testing.html#building-matplotlib-for-image-comparison-tests

**Comment by three-comrades** [comment]
Okay, I've recreated the test with the built-in FreeType version.

**Comment by tacaswell** [comment]
'power cycling' this PR to make it rebuild against current master.

**Comment by three-comrades** [comment]
Rebased. Last time a Python 2.7 test failed that seemed unrelated to me (matplotlib.tests.test_mathtext.test_mathfont_stix_07.test). Let's see if that is still the case.

File: lib/matplotlib/legend_handler.py [hunk 1]
```diff
@@ -565,9 +568,29 @@ def create_artists(self, legend, orig_handle,
...
-    Handler for Tuple
+    Handler for Tuple.
+
+    Additional kwargs are passed through to `HandlerBase`.
+
+    Parameters
+    ----------
+
+    ndivide : int, optional
+        The number of sections to divide the legend area into.  If 0,
```
Comment (tacaswell) [inline lib/matplotlib/legend_handler.py]: Can we use `None` as the sentinel value?

Comment (three-comrades) [inline lib/matplotlib/legend_handler.py]: Sure.

**Comment by WeatherGod** [comment]
I think this needs a whats_new entry?

**Comment by three-comrades** [comment]
I put a description in `doc/users/whats_new/multiple_legend_keys.rst`. Is that enough?

**Comment by tacaswell** [comment]
@three-comrades Yes, that looks sufficient to me.

**Comment by tacaswell** [comment]
:+1: from me, but I will refrain from pushing the green button because I have commits in this PR.

**Pull request merged**